xgi/generators/uniform.py

```diff
@@ -346,6 +369,47 @@ def _index_to_edge(index, n, m):
     return [(index // (n**r) % n) for r in range(m - 1, -1, -1)]
 
 
+def _index_to_edge_comb(index, n, m):
+    """Generate a hyperedge given an index in the list of possible edges.
+
+    In this function, we prohibit multiedges, so each index corresponds to a
+    unique edge.
+
+    Parameters
+    ----------
+    index : int > 0
+        The index of the hyperedge in the list of all possible hyperedges.
+    n : int > 0
+        The number of nodes
+    m : int > 0
+        The hyperedge size.
+
+    Returns
+    -------
+    list
+        The reconstructed hyperedge
+
+    See Also
+    --------
+    _index_to_edge_partition
+
+    References
+    ----------
+    https://math.stackexchange.com/questions/1227409/indexing-all-combinations-without-making-list
+    """
+    c = []
+    r = index
+    j = -1
+    for s in range(1, m + 1):
+        cs = j + 1
+        while r - comb(n - 1 - cs, m - s, exact=True) > 0:
+            r -= comb(n - 1 - cs, m - s, exact=True)
+            cs += 1
+        c.append(cs)
+        j = cs
+    return c
+
+
 def _index_to_edge_partition(index, partition_sizes, m):
     """Generate a hyperedge given an index in the list of possible edges
```

Comment:
given those things, how is it generated?

Reply:
Fixed.